src/textual/css/_styles_builder.py

```diff
@@ -63,27 +68,37 @@ def add_declaration(self, declaration: Declaration) -> None:
         if not declaration.tokens:
             return
         rule_name = declaration.name.replace("-", "_")
-        process_method = getattr(self, f"process_{rule_name}", None)
-
-        if process_method is None:
+        if rule_name not in self._processable_rule_names():
+            error_message = f"unknown declaration {declaration.name!r}"
+            did_you_mean_rule_name = self._did_you_mean_for_rule_name(declaration.name)
+            if did_you_mean_rule_name:
+                error_message += f"; did you mean {did_you_mean_rule_name!r}?"
             self.error(
                 declaration.name,
                 declaration.token,
-                f"unknown declaration {declaration.name!r}",
+                error_message,
             )
-        else:
-            tokens = declaration.tokens
+            return
 
-            important = tokens[-1].name == "important"
-            if important:
-                tokens = tokens[:-1]
-                self.styles.important.add(rule_name)
-            try:
-                process_method(declaration.name, tokens)
-            except DeclarationError:
-                raise
-            except Exception as error:
-                self.error(declaration.name, declaration.token, str(error))
+        process_method: CssPropertyProcessMethod = getattr(self, f"process_{rule_name}")
+        tokens = declaration.tokens
+
+        important = tokens[-1].name == "important"
+        if important:
+            tokens = tokens[:-1]
+            self.styles.important.add(rule_name)
+        try:
+            process_method(declaration.name, tokens)
+        except DeclarationError:
+            raise
+        except Exception as error:
+            self.error(declaration.name, declaration.token, str(error))
+
+    @lru_cache(maxsize=None)
+    def _processable_rule_names(self) -> frozenset[str]:
+        return frozenset(
```

Comment:
I suspect it doesn't need to be a set. Correct me if I'm wrong but we will only ever iterate through the container, and never check for set membership. In that case, a plain old list is probably better.

Reply:
It doesn't need to be a set indeed - I chose this data structure only because it seems to be the one with the fastest lookup. But as this is something that is only triggered when fatal errors happen I guess performance is not really a concern there.
I'll change that to a list! :slightly_smiling_face:  :+1: